File: controllers/inventory/inventory.controller.js
import InventoryItem from '../../models/inventory/InventoryItem.js';
import StockTransaction from '../../models/inventory/StockTransaction.js';

export const createInventoryItem = async (req, res) => {
    try {
        const item = new InventoryItem(req.body);
        await item.save();
        res.status(201).send(item);
    } catch (error) {
        res.status(400).send(error);
    }
};

export const getInventoryItems = async (req, res) => {
    try {
        const items = await InventoryItem.find({});
        res.send(items);
    } catch (error) {
        res.status(500).send(error);
    }
};

export const createStockTransaction = async (req, res) => {
    try {
        const transaction = new StockTransaction(req.body);
        await transaction.save();
        // Update inventory item quantity
        const item = await InventoryItem.findById(transaction.item);
        if (transaction.type === 'in') {
            item.quantity += transaction.quantity;
        } else {
            item.quantity -= transaction.quantity;
        }
        await item.save();
        res.status(201).send(transaction);
    } catch (error) {
        res.status(400).send(error);
    }
};

File: jstojv/src/main/java/com/samrat/jstojv/controller/InventoryController.java
package com.samrat.jstojv.controller;

import com.samrat.jstojv.entity.InventoryItem;
import com.samrat.jstojv.entity.StockTransaction;
import com.samrat.jstojv.service.InventoryManagementService;
import lombok.RequiredArgsConstructor;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@RestController
@RequestMapping("/api/v1/inventory")
@RequiredArgsConstructor
public class InventoryController {

    private final InventoryManagementService inventoryManagementService;

    @PostMapping("/items")
    public ResponseEntity<InventoryItem> createInventoryItem(@RequestBody InventoryItem item) {
        return new ResponseEntity<>(inventoryManagementService.createInventoryItem(item), HttpStatus.CREATED);
    }

    @GetMapping("/items")
    public ResponseEntity<List<InventoryItem>> getInventoryItems() {
        return ResponseEntity.ok(inventoryManagementService.getAllInventoryItems());
    }

    @PostMapping("/transactions")
    public ResponseEntity<StockTransaction> createStockTransaction(@RequestBody StockTransaction transaction) {
        return new ResponseEntity<>(inventoryManagementService.createStockTransaction(transaction), HttpStatus.CREATED);
    }
}
